src/renderer/ssf-api.ts

```diff
@@ -236,6 +237,28 @@ export class SSFApi {
         }
     }
 
+    /**
+     * Allows JS to register a log retriever that can be used by electron main process.
+     * @param  {Object} retriever  function that can be called to retrieve logs
+     */
+    public registerLogRetriever(retriever): void {
+        if (typeof retriever === 'function') {
+            local.logRetriever = retriever;
```

Comment:
What if you register two log retrievers?

Reply:
I don't think there is a use-case for that though. But ok, what change is needed here?

Reply:
Yes there is. We want to use this interface from RTC. If some other app later want to add some logs, for example SFE-Lite or the email extension, that will not work.

If we can only have a single log receiver, we need to document that properly, and we need to write the code to collect logs from multiple sources somewhere else. But I think this would be the best place to do that.

Reply:
When sending the logs, it is a list of log files, so we could send multiple logs (gathered from multiple sources if needed) whenever the retriever is invoked.

Reply:
Yes, but there would have to be some code in, for example, SFE-Lite that would gather the log files from different sources in the app. And the same code would have to be put in SFE-Client-App.

It would be better to have this API support multiple users within the app. That's usually how you do browser APIs.

Reply:
Ok, so what would you suggest we do? How should the API look?